src/Runner.Worker/ExecutionContext.cs

```diff
@@ -52,8 +52,7 @@ public interface IExecutionContext : IRunnerService
         Dictionary<string, string> IntraActionState { get; }
         Dictionary<string, VariableValue> JobOutputs { get; }
         ActionsEnvironmentReference ActionsEnvironment { get; }
-        List<ActionsStepTelemetry> ActionsStepsTelemetry { get; }
-        List<JobTelemetry> JobTelemetry { get; }
+        ActionsStepTelemetry StepTelemetry { get; }
```

Comment:
Each `ExecutionContext` will hold a single `ActionsStepTelemetry` object, the object will get append to the `GlobalContext.ActionsStepsTelemetry` when `ExecutionContext.Complete()` is called.

Reply:
For composite child steps we don't call `complete()` so I don't think you will be getting that data currently

Reply:
thanks for pointing this out, i will take a look.

Reply:
I am adding a new `PublishStepTelemetry()` to executionContext, so composite actions can use that to publish step telemetry.